pkg/workloadmeta/collectors/internal/docker/image_sbom_trivy.go

```diff
@@ -82,6 +82,10 @@ func (c *collector) startSBOMCollection(ctx context.Context) error {
 
 	go func() {
 		for result := range resultChan {
+			if result.ImgMeta == nil {
+				log.Errorf("Scan result does not hold the image identifier. Error: %s", result.Error)
+				continue
+			}
```

Comment:
IMO, `ImgMeta` should always be available because we get it from the request. so it should always be possible to copy the imgMeta from the request to the result.

Reply:
> This PR fix will avoid to have the previous agent panic. But it doesn't seems to resolve the initial issues which is that the result.ImgMeta is nil, which should be resolved because the image metadata info should be available in the request, so even in case of a scan error, the image metadata can be copy from the request to the result.

It should fix the original issue. If the `imgMeta` is available we retrieve it with 
https://github.com/DataDog/datadog-agent/blob/6010b11ee3815773f04492c6d03496b609ca38c7/pkg/sbom/scanner/scanner.go#L120-L127

As an example, it logs the error in `workload-list -v` with the associated image.
```
----------- SBOM -----------
Status: Failed
Error: failed to check current disk usage: not enough disk space to safely collect sbom, 192108482560 available, 1073741824000 required
```

Reply:
> IMO, `ImgMeta` should always be available because we get it from the request. so it should always be possible to copy the imgMeta from the request to the result.

Indeed, I added this as an additional safeguard thinking it could help if we miss an edge case `imgMeta`. For example there is this one, which I never observed in practice:
https://github.com/DataDog/datadog-agent/blob/65711d918a30f2d0e2baff9abeeb9da3b40444ff/pkg/sbom/collectors/docker/docker.go#L74